### ISSUE TITLE
Named parameters for screed.Record constructor

### ISSUE THREAD

**Issue by betatim** [dib-lab/khmer#1498]
The `screed` and `khmer` classes to represent a read have nearly the same interface and behaviour but not quite.

As part of https://github.com/dib-lab/khmer/pull/1492#issuecomment-257236074 we decided to make `screed.Record` support a constructor like `khmer.Read`. Namely named parameters instead of passing a dictionary as only argument.

**Comment by luizirber** [comment]
This already works, no?

```python
>>> import screed
>>> a = screed.Record(name="seq1", sequence="ACGTC", quality="NNNNN")
>>> a
{'sequence': 'ACGTC', 'name': 'seq1', 'quality': 'NNNNN'}
```

(this is a side effect of how we read the dict to create the record)

```python
def __init__(self, *args, **kwargs):                                                   
    self.d = dict(*args, **kwargs) 
```

The question then becomes: do we want to accept anything, or limit to the same interface as `khmer.Read`?

**Comment by standage** [comment]
- Limiting the interface will be a little bit more code (= maintenance burden).
- Not limiting the interface makes accidental inits more likely (`screed.Record(seq='ACGT')` instead of `screed.Record(sequence='ACGT')`), which could lead to subtle bugs.

I'm not sure we have a good intuition for which of these is going to be more costly to us in the long term, so I'm in favor of just keeping things as they are: not explicitly restricting the interface to some set of supported dict keys, but making sure it works correctly when `quality` is or is not defined.

**Comment by ctb** [comment]
On Mon, Dec 19, 2016 at 11:41:34AM -0800, Daniel Standage wrote:
> - Limiting the interface will be a little bit more code (= maintenance burden).
> - Not limiting the interface makes accidental inits more likely (`screed.Record(seq='ACGT')` instead of `screed.Record(sequence='ACGT')`), which could lead to subtle bugs.
> 
> I'm not sure we have a good intuition for which of these is going to be more costly to us in the long term, so I'm in favor of just keeping things as they are: not explicitly restricting the interface to some set of supported dict keys, but making sure it works correctly when `quality` is or is not defined.

I'm pretty sure the latter is more problematic; we need to require a name
and sequence at a minimum, IMO.

see https://github.com/dib-lab/screed/pull/54

**Comment by luizirber** [comment]
fixed by dib-lab/screed#54
